Library/Homebrew/utils.rb

```diff
@@ -337,6 +337,14 @@ def with_system_path
   ENV["PATH"] = old_path
 end
 
+def with_custom_locale(locale)
+  old_locale = ENV["LC_ALL"]
+  ENV["LC_ALL"] = locale
+  yield
+ensure
+  ENV["LC_ALL"] = old_locale
+end
```

Comment:
Two cents: I wrote the testing code ;) but I think it makes sense in the main codebase and I like that it's more explicit than `with_custom_locale`. I'm happy either way.